src/chat_sdk/conversations/private_v1.nim

```diff
@@ -94,6 +94,10 @@ proc decrypt*(convo: PrivateV1, enc: EncryptedPayload): Result[seq[byte], ChatEr
   )
   copyMem(addr header.dhPublic[0], unsafeAddr dr.dh[0], dr.dh.len) # TODO: Avoid this copy
 
+  if convo.doubleratchet.dhSelf.public == header.dhPublic:
+    info "outgoing message, no need to decrypt"
+    return err(ChatError(code: errDecryptOutgoing, context: "Attempted to decrypt outgoing message"))
+
```

Comment:
Returning an error here means that handleFrame will fail with a decryption error. 

I'd suggest that handleFrame ought to return early in this case rather than failing.